python/version_details/shotgun_entities/card_delegate.py

```diff
@@ -167,11 +171,8 @@ def _get_painter_widget(self, model_index, parent):
         if model_index in self._widget_cache:
             widget = self._widget_cache[model_index]
 
-            if self.fields != widget.fields:
-                widget.fields = self.fields
-                self.sizeHintChanged.emit(model_index)
-
-            return widget
+            if sorted(self.fields) == sorted(widget.fields):
```

Comment:
Yeah, I agree that this is unfortunate. However, ordering is important in most cases in this chain of widgets, os simply using a set to store the list of fields wouldn't be appropriate in other situations where that data is utilized.